subgraphs/euler-finance/src/mappings/handlers.ts

```diff
@@ -293,3 +324,110 @@ export function handleGovSetReserveFee(event: GovSetReserveFee): void {
+  // find market id for underlying
+  const assetStatus = getOrCreateAssetStatus(underlying);
+  const marketId = assetStatus.eToken!;
+  const market = getOrCreateMarket(marketId);
+  const deltaStakedAmount = getDeltaStakeAmount(event);
+
+  const epochID = getCurrentEpoch(event);
+  if (epochID < 0) {
+    market._stakedAmount = market._stakedAmount.plus(deltaStakedAmount);
+    market.save();
+    return;
+  }
+
+  const epochStartBlock = getStartBlockForEpoch(epochID)!;
+  let epoch = _Epoch.load(epochID.toString());
+  if (!epoch) {
+    //Start of a new epoch
+    epoch = new _Epoch(epochID.toString());
+    epoch.epoch = epochID;
+    epoch.save();
+
+    // rank markets use votes in prev epoch
+    // find the top ten staked markets; according to the euler guage
+    // https://app.euler.finance/gaugeweight
+    // users "vote" for the next epoch
+    const prevEpochID = epochID - 1;
+    const prevEpoch = _Epoch.load(prevEpochID.toString());
+
+    const protocol = getOrCreateLendingProtocol();
+    if (prevEpoch) {
+      // finalize mkt._weightedStakedAmount and mrkt._weightedTotalBorrowUSD
+      // epoch.top10StakeAmounts for prev Epoch
+      const marketStakeAmounts: BigInt[] = [];
+      let sumWeightedBorrowUSD = BIGDECIMAL_ZERO;
+      for (let i = 0; i < protocol._marketIDs!.length; i++) {
+        const mktID = protocol._marketIDs![i];
+        const mrkt = getOrCreateMarket(mktID);
+        const stakedAmount = mrkt._stakedAmount;
+        if (stakedAmount.gt(BIGINT_ZERO)) {
+          updateWeightedStakeAmount(mrkt, epochStartBlock);
+          mrkt._stakeLastUpdateBlock = epochStartBlock;
+        }
+        marketStakeAmounts.push(mrkt._weightedStakedAmount ? mrkt._weightedStakedAmount! : BIGINT_ZERO);
+        if (mrkt.totalBorrowBalanceUSD.gt(BIGDECIMAL_ZERO)) {
+          updateWeightedBorrow(mrkt, prevEpoch, epochStartBlock);
+        }
+        sumWeightedBorrowUSD = sumWeightedBorrowUSD.plus(
+          mrkt._weightedTotalBorrowUSD ? mrkt._weightedTotalBorrowUSD! : BIGDECIMAL_ZERO,
+        );
+        mrkt.save();
+      }
+
+      const eulerContract = Euler.bind(Address.fromString(EULER_ADDRESS));
+      const execProxyAddress = eulerContract.moduleIdToProxy(MODULEID__EXEC);
+      // update EUL token prices
+      updatePrices(execProxyAddress, market, event);
+
+      const cutoffAmount = getCutoffValue(marketStakeAmounts, 10);
+      const totalRewardAmount = BigDecimal.fromString((EUL_DIST[prevEpochID - START_EPOCH] * EUL_DECIMALS).toString());
+
+      const EULToken = getOrCreateToken(Address.fromString(EUL_ADDRESS));
+      const rewardToken = getOrCreateRewardToken(Address.fromString(EUL_ADDRESS), RewardTokenType.BORROW);
+      const dailyScaler = BigDecimal.fromString((BLOCKS_PER_DAY / (BLOCKS_PER_EPOCH as f64)).toString());
+      for (let i = 0; i < protocol._marketIDs!.length; i++) {
+        const mktID = protocol._marketIDs![i];
+        const mrkt = getOrCreateMarket(mktID);
```

Comment:
Kinda the same idea with what you mentioned on my PR. My take is that it is supposed to be there, so if it is not you would want to throw an error and skip

Reply:
To clarify, are you referring to `mrkt = getOrCreateMarket(mktID)` or `if (!mrktRewardToken || mrktRewardToken.length == 0)`. I think, in my case, `getOrCreateMarket` always returns a valid market, and `rewardToken = getOrCreateRewardToken()` should alwasys returns a valid rewardToken?

Reply:
`mrkt = getOrCreateMarket(mktID)` is what I am talking about.

And I agree it will always return a Market entity. I guess my point is that if it had to make a Market entity that would be an error. It really shouldn't happen.

I usually do something like this:
```
const mrkt = Market.load(mktID);
if (!mrkt) {
     log.error();
     return;
}
```

Idk it rly isn't a major thing, I'm just being picky and we are really just informally defining standards as we do this

Reply:
OK, makes sense. I am switching to load & check.